modules/apps/address/address-test/src/testIntegration/java/com/liferay/address/service/test/RegionLocalServiceTest.java

```diff
@@ -0,0 +1,141 @@
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testAddRegion() throws Exception {
+		boolean active = RandomTestUtil.randomBoolean();
+		String name = RandomTestUtil.randomString();
+		double position = RandomTestUtil.randomDouble();
+		String regionCode = RandomTestUtil.randomString();
+
+		Country country = _addCountry();
+
+		Region region = _addRegion(
+			country.getCountryId(), active, name, position, regionCode,
+			ServiceContextTestUtil.getServiceContext());
+
+		Assert.assertNotNull(
+			_regionLocalService.fetchRegion(region.getRegionId()));
+
+		User user = TestPropsValues.getUser();
+
+		Address address = _addressLocalService.addAddress(
+			null, user.getUserId(), null, user.getContactId(),
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), null, null,
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			region.getRegionId(), region.getCountryId(),
+			RandomTestUtil.randomLong(), false, false, "1234567890",
+			ServiceContextTestUtil.getServiceContext());
+
+		Assert.assertEquals(region.getCountryId(), address.getCountryId());
+		Assert.assertEquals(region.getRegionId(), address.getRegionId());
+
+		_regionLocalService.deleteRegion(region.getRegionId());
+
+		Assert.assertNull(
+			_addressLocalService.fetchAddress(address.getAddressId()));
+		Assert.assertNull(
+			_regionLocalService.fetchRegion(region.getRegionId()));
+	}
+
+	@Test
+	public void testUpdateRegion() throws Exception {
+		Country country = _addCountry();
+
+		Region region = _addRegion(
+			country.getCountryId(), RandomTestUtil.randomBoolean(),
+			RandomTestUtil.randomString(), RandomTestUtil.randomDouble(),
+			RandomTestUtil.randomString(),
+			ServiceContextTestUtil.getServiceContext());
+
+		boolean active = RandomTestUtil.randomBoolean();
+		String name = RandomTestUtil.randomString();
+		double position = RandomTestUtil.randomDouble();
+		String regionCode = RandomTestUtil.randomString();
+
+		region = _regionLocalService.updateRegion(
+			region.getRegionId(), active, name, position, regionCode);
+
+		Assert.assertEquals(active, region.isActive());
+		Assert.assertEquals(name, region.getName());
+		Assert.assertEquals(position, region.getPosition(), 0);
+		Assert.assertEquals(regionCode, region.getRegionCode());
+	}
+
+	private Country _addCountry() throws Exception {
+		return _countryLocalService.addCountry(
+			RandomTestUtil.randomString(2), RandomTestUtil.randomString(3),
+			true, RandomTestUtil.randomBoolean(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomDouble(), RandomTestUtil.randomBoolean(),
+			RandomTestUtil.randomBoolean(), RandomTestUtil.randomBoolean(),
+			ServiceContextTestUtil.getServiceContext());
+	}
+
+	private Region _addRegion(
+			long countryId, boolean active, String name, double position,
+			String regionCode, ServiceContext serviceContext)
```

Comment:
We can simplify this by just asking for the `countryId` for the param. All other params can be random values.